sql/core/src/main/scala/org/apache/spark/sql/execution/python/TransformWithStateInPandasStateServer.scala

```diff
@@ -158,6 +229,51 @@ class TransformWithStateInPandasStateServer(
           Some(message.getGetValueState.getTtl.getDurationMs)
         } else None
         initializeValueState(stateName, schema, ttlDurationMs)
+
+      case StatefulProcessorCall.MethodCase.TIMERSTATECALL =>
+        message.getTimerStateCall.getMethodCase match {
+          case TimerStateCallCommand.MethodCase.REGISTER =>
+            val expiryTimestamp =
+              message.getTimerStateCall.getRegister.getExpiryTimestampMs
+            statefulProcessorHandle.registerTimer(expiryTimestamp)
+            sendResponse(0)
+          case TimerStateCallCommand.MethodCase.DELETE =>
+            val expiryTimestamp =
+              message.getTimerStateCall.getDelete.getExpiryTimestampMs
+            statefulProcessorHandle.deleteTimer(expiryTimestamp)
+            sendResponse(0)
+          case TimerStateCallCommand.MethodCase.LIST =>
+            val iter = statefulProcessorHandle.listTimers()
+
+            if (iter == null || !iter.hasNext) {
+              // avoid sending over empty batch
+              sendResponse(1)
+            } else {
+              sendResponse(0)
+              outputStream.flush()
+              val arrowStreamWriter = {
+                val outputSchema = new StructType()
+                  .add(StructField("timestamp", LongType))
+                val arrowSchema = ArrowUtils.toArrowSchema(outputSchema, timeZoneId,
+                  errorOnDuplicatedFieldNames, largeVarTypes)
+                val allocator = ArrowUtils.rootAllocator.newChildAllocator(
+                  s"stdout writer for transformWithStateInPandas state socket", 0, Long.MaxValue)
+                val root = VectorSchemaRoot.create(arrowSchema, allocator)
+                new BaseStreamingArrowWriter(root, new ArrowStreamWriter(root, null, outputStream),
+                  arrowTransformWithStateInPandasMaxRecordsPerBatch)
+              }
+              while (iter.hasNext) {
+                val timestamp = iter.next()
+                val internalRow = InternalRow(timestamp)
+                arrowStreamWriter.writeRow(internalRow)
```

Comment:
Same as the python side comment: here we don't limit how many arrow batches we construct for timers, if user sets a fairly low value for arrowTransformWithStateInPandasMaxRecordsPerBatch, we would send multiple arrow batches and client side needs to handle this properly as well.

Question: should we have a lower limit on how many records we send throw a single batch (e.g. the default value 10000)? IIUC, each timer record is very small and should not consume a lot of memory. The user also doesn't care about how many records each batch contains since they would always get a single list from this API.